pkg/operator/scheduled_reports.go

```diff
@@ -503,10 +513,21 @@ func (op *Reporting) runScheduledReport(logger log.FieldLogger, report *cbTypes.
 	return nil
 }
 
-func getNextReportPeriod(schedule reportSchedule, period cbTypes.ScheduledReportPeriod, lastScheduled time.Time) reportPeriod {
+func getRunOnceReportPeriod(report *cbTypes.ScheduledReport) (*reportPeriod, error) {
+	if report.Spec.ReportingEnd == nil || report.Spec.ReportingStart == nil {
```

Comment:
@tschuy, I think we should add more error checking on validity of start and end time.

Reply:
the ReportingEnd and ReportingStart are compared to ensure they're in the right order upon running the report: https://github.com/operator-framework/operator-metering/blob/master/pkg/operator/scheduled_reports.go#L212-L218

We could duplicate this check here as well. That would mean the actual report itself fails to validate rather than the run of it. @chancez thoughts on this?

Reply:
We're planning on supporting reports without a reportingStart/reportingEnd, so I don't think it makes sense. We just need to make our ReportGenerationQuery spec.query templates handle it and we can actually remove the validation.

Reply:
What would happen if the user gave a reportingEnd time that is earlier than reportingStart?